From 10dcc76db32d36cdfaf9a33aed56696c9553a1af Mon Sep 17 00:00:00 2001
From: Christian Kastner <ckk@kvr.at>
Date: Fri, 1 May 2020 21:10:50 +0200
Subject: [PATCH] autopkgtest-build-qemu: Support for vmdb2->qemu-debootstrap

This makes it possible to build i386 images on an amd64 host, for
example.

This only works for architectures that use BIOS GRUB. To extend this to
other architectures, the vmdb2 configuration would need to be extended
again.
---
 tools/autopkgtest-build-qemu | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/tools/autopkgtest-build-qemu b/tools/autopkgtest-build-qemu
index 16fdc2a..6a68ba8 100755
--- a/tools/autopkgtest-build-qemu
+++ b/tools/autopkgtest-build-qemu
@@ -235,6 +235,16 @@ case "$mirror" in
     ;;
 esac
 
+
+if [ "$architecture" = "$(dpkg --print-architecture)" ]; then
+    debootstrap_cmd=debootstrap
+    debootstrap_arch=
+else
+    debootstrap_cmd=qemu-debootstrap
+    debootstrap_arch="arch: $architecture"
+fi
+
+
 vmdb2_config=$(mktemp)
 trap "rm -rf $vmdb2_config" INT TERM EXIT
 cat > "$vmdb2_config" <<EOF
@@ -258,9 +268,10 @@ steps:
 
   - mount: root
 
-  - debootstrap: $release
+  - $debootstrap_cmd: $release
     mirror: $mirror
     target: root
+    $debootstrap_arch
 
   - apt: install
     packages:
-- 
2.26.2

